pyanaconda/storage/osinstall.py

```diff
@@ -0,0 +1,2161 @@
+             self.config.clear_part_type.
+
+             :keyword disks: overrides :attr:`disks`
+             :type disks: list
+             :keyword clear_part_type: overrides :attr:`self.config.clear_part_type`
+             :type clear_part_type: int
+             :returns: dict with disk name keys and tuple (disk, fs) free values
+             :rtype: dict
+
+            .. note::
+
+                The free space values are :class:`blivet.size.Size` instances.
+
+        """
+
+        # FIXME: we should definitely do something with this method -- it takes
+        # different parameters than get_free_space from Blivet and does
+        # different things too
+
+        if disks is None:
+            disks = self.disks
+
+        if clear_part_type is None:
+            clear_part_type = self.config.clear_part_type
+
+        free = {}
+        for disk in disks:
+            should_clear = self.should_clear(disk, clear_part_type=clear_part_type,
+                                             clear_part_disks=[disk.name])
+            if should_clear:
+                free[disk.name] = (disk.size, Size(0))
+                continue
+
+            disk_free = Size(0)
+            fs_free = Size(0)
+            if disk.partitioned:
+                disk_free = disk.format.free
+                for partition in (p for p in self.partitions if p.disk == disk):
+                    # only check actual filesystems since lvm &c require a bunch of
+                    # operations to translate free filesystem space into free disk
+                    # space
+                    should_clear = self.should_clear(partition,
+                                                     clear_part_type=clear_part_type,
+                                                     clear_part_disks=[disk.name])
+                    if should_clear:
+                        disk_free += partition.size
+                    elif hasattr(partition.format, "free"):
+                        fs_free += partition.format.free
+            elif hasattr(disk.format, "free"):
+                fs_free = disk.format.free
+            elif disk.format.type is None:
+                disk_free = disk.size
+
+            free[disk.name] = (disk_free, fs_free)
+
+        return free
+
+    def update_ksdata(self):
+        """ Update ksdata to reflect the settings of this Blivet instance. """
+        if not self.ksdata or not self.mountpoints:
+            return
+
+        # clear out whatever was there before
+        self.ksdata.partition.partitions = []
+        self.ksdata.logvol.lvList = []
+        self.ksdata.raid.raidList = []
+        self.ksdata.volgroup.vgList = []
+        self.ksdata.btrfs.btrfsList = []
+
+        # iscsi?
+        # fcoe?
+        # zfcp?
+        # dmraid?
+
+        # bootloader
+
+        # ignoredisk
+        if self.ignored_disks:
+            self.ksdata.ignoredisk.drives = self.ignored_disks[:]
```

Comment:
It should be `self.ksdata.ignoredisk.ignoredisk`.

Reply:
Will fix -- thanks.